Core/src/com/infiniteautomation/mango/spring/service/UserCommentService.java

```diff
@@ -157,4 +178,11 @@ else if (!isXidUnique(vo.getXid(), vo.getId()))
         return result;
     }
 
+    private void reloadEventCache(UserCommentVO vo) {
+        if (vo.getCommentType() == UserCommentVO.TYPE_EVENT) {
+            int eventId = vo.getReferenceId();
+            Common.eventManager.reloadEvent(eventId);
```

Comment:
It might be more performant to manage the in-memory comment lists doing something like this instead of reloading the event from the DB.  However you will need to manipulate the comment list on the event in a thread safe way because someone could be looping through that list or another thread could be performing the modification at the same time.  I would do this by Copy On Write logic such that you duplicate the existing list, modify the copy and then set the copy into the EventInstance as a last step.

```
private void manageEventComments(UserCommentVO vo, DaoEventType action) {
    if (vo.getCommentType() == UserCommentVO.TYPE_EVENT) {
        EventInstance evt = Common.eventManager.getById(vo.getReferenceId());
        if(evt!=null) { 
            switch(action) {
                case CREATE:
                //TODO add via evt.setEventComments()
               break;
               case UPDATE:
              // TODO replace comment in event comments on event
               break;
               case DELETE:
               // Not sure if this is possible, but remove from list.
               break;
        }
    }
    
}
```

Reply:
An approach like this would prevent adding a method to the EventManager